From 87beeb324aa4c8611f8da47ac4d75a8b8c4c865f Mon Sep 17 00:00:00 2001
From: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Date: Tue, 28 Nov 2017 13:33:32 +0900
Subject: [PATCH 379/909] media: i2c: adv748x: Add g_std callback function for
 csi2

commit 76ad1cc100b43e36f189a842da4c3cbe9b34d3e2 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

Since it is necessary to acquire field information with other
modules, register a callback function for acquiring field
information in adv748x driver.

Signed-off-by: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/media/i2c/adv748x/adv748x-csi2.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/drivers/media/i2c/adv748x/adv748x-csi2.c b/drivers/media/i2c/adv748x/adv748x-csi2.c
index de30d489b60d..d16eec175693 100644
--- a/drivers/media/i2c/adv748x/adv748x-csi2.c
+++ b/drivers/media/i2c/adv748x/adv748x-csi2.c
@@ -106,6 +106,18 @@ static const struct v4l2_subdev_internal_ops adv748x_csi2_internal_ops = {
  * v4l2_subdev_video_ops
  */
 
+static int adv748x_csi2_g_std(struct v4l2_subdev *sd, v4l2_std_id *std)
+{
+	struct adv748x_csi2 *tx = adv748x_sd_to_csi2(sd);
+	struct v4l2_subdev *src;
+
+	src = adv748x_get_remote_sd(&tx->pads[ADV748X_CSI2_SINK]);
+	if (!src)
+		return -EPIPE;
+
+	return v4l2_subdev_call(src, video, g_std, std);
+}
+
 static int adv748x_csi2_s_stream(struct v4l2_subdev *sd, int enable)
 {
 	struct adv748x_csi2 *tx = adv748x_sd_to_csi2(sd);
@@ -119,6 +131,7 @@ static int adv748x_csi2_s_stream(struct v4l2_subdev *sd, int enable)
 }
 
 static const struct v4l2_subdev_video_ops adv748x_csi2_video_ops = {
+	.g_std = adv748x_csi2_g_std,
 	.s_stream = adv748x_csi2_s_stream,
 };
 
-- 
2.17.1

